docs/linters.md

```diff
@@ -383,6 +384,168 @@ lintersConfig:
+
+```go
+// ✅ Valid: marker on type definition
+// +kubebuilder:validation:Minimum=0
+type Port int32
+
+type Service struct {
+    Port Port `json:"port"`
+}
+
+// ❌ Invalid: marker on field using named type
+type Port int32
+
+type Service struct {
+    // +kubebuilder:validation:Minimum=0 // Error: should be on Port type definition
+    Port Port `json:"port"`
+}
+```
+
+Most built-in kubebuilder validation markers use `strictTypeConstraint: true` to encourage consistent marker placement on type definitions.
+
+### Default Marker Rules
+
+The linter includes built-in rules for all standard kubebuilder markers and k8s declarative validation markers. Examples:
+
+**Field-only markers:**
+- `optional`, `required`, `nullable`
+- `kubebuilder:default`, `kubebuilder:validation:Example`
+
+**Type-only markers:**
+- `kubebuilder:validation:items:ExactlyOneOf`
+- `kubebuilder:validation:items:AtMostOneOf`
+- `kubebuilder:validation:items:AtLeastOneOf`
+
+**Field and Type markers with type constraints:**
+- `kubebuilder:validation:Minimum` (integer types only)
+- `kubebuilder:validation:Pattern` (string types only)
+- `kubebuilder:validation:MinItems` (array types only)
+- `kubebuilder:validation:MinProperties` (object types only)
+
+**Field and Type markers without type constraints:**
+- `kubebuilder:validation:Enum`, `kubebuilder:validation:Format`
+- `kubebuilder:pruning:PreserveUnknownFields`, `kubebuilder:title`
+
+### Configuration
+
+You can customize marker rules or add support for custom markers.
+
+**Scope values:**
+
+The `scopes` field accepts an array of scope constraints:
+- `[Field]`: Marker can only be applied to struct fields
+- `[Type]`: Marker can only be applied to type definitions
+- `[Field, Type]`: Marker can be applied to both fields and type definitions
+
+**Type constraints:**
+
+The `typeConstraint` field allows you to restrict which Go types a marker can be applied to. This ensures that markers are only used with compatible data types (e.g., numeric markers like `Minimum` are only applied to integer types).
+
+**Type constraint fields:**
+- `allowedSchemaTypes`: List of allowed OpenAPI schema types (`integer`, `string`, `boolean`, `array`, `object`)
+- `elementConstraint`: Nested constraint for array element types (only valid when `allowedSchemaTypes` includes `array`)
+- `strictTypeConstraint`: When `true`, markers that can be applied to both fields and types with type constraints applied to fields using named types must be declared on the type definition instead of the field. Defaults to `false`.
+
+**Configuration example:**
+
+```yaml
+lintersConfig:
+  markerscope:
+    policy: Warn | SuggestFix # The policy for marker scope violations. Defaults to `Warn`.
+
+    # Override default rules for built-in markers
+    overrideMarkers:
+      - identifier: "optional"
+        scopes: [Field]  # Can specify [Field], [Type], or [Field, Type]
+
+    # Add rules for custom markers
+    customMarkers:
```

Comment:
Do we need separate lists here? Or shall we just assume anything in the custom list that matches a built-in config overrides it?